docs/src/_contribute/plugin-definition-syntax.md

```diff
@@ -0,0 +1,332 @@
+```
+
+## `settings_preamble`
+
+Text to display before the settings in Meltano Hub, in Markdown format.
+
+```yaml
+settings_preamble: |
+  Meltano [centralizes the configuration](https://docs.meltano.com/guide/configuration) of all of the plugins in your project, including Airflow's. This means that if the [Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) tells you to put something in `airflow.cfg`, you can use `meltano config`, `meltano.yml`, or environment variables instead, and get the benefits of Meltano features like [environments](https://docs.meltano.com/concepts/environments).
+
+  Any setting you can add to `airflow.cfg` can be added to `meltano.yml`, manually or using `meltano config`. For example, `[core] executor = SequentialExecutor` becomes `meltano config airflow set core executor SequentialExecutor` on the CLI, or `core.executor: SequentialExecutor` in `meltano.yml`. Config sections indicated by `[section]` in `airflow.cfg` become nested dictionaries in `meltano.yml`.
+```
+
+## `next_steps`
+
+Documentation for the next steps after installing the plugin, in Markdown format.
+
+```yaml
+next_steps: |
+  1. Use the [meltano schedule](https://docs.meltano.com/reference/command-line-interface#schedule) command to create pipeline schedules in your project, to be run by Airflow.
+
+  1. If you're running Airflow for the first time in a new environment, create an admin user:
+
+     ```sh
+     meltano invoke airflow:create-admin
+     # This is equivalent to `airflow users create` with some arguments in the Airflow documentation
+     ```
+
+  1. Launch the Airflow UI and log in using the username/password you created:
+
+     ```sh
+     meltano invoke airflow:ui
+     ```
+
+     By default, the UI will be available at at [`http://localhost:8080`](http://localhost:8080). You can change this using the `webserver.web_server_port` setting documented below.
+
+  1. Start Scheduler or execute Airflow commands directly using the instructions in [the Meltano docs](https://docs.meltano.com/guide/orchestration#starting-the-airflow-scheduler).
+```
+
+## `usage``
+
+Free text describing how to use the plugin, in Markdown format.
+
+```yaml
+usage: |
+  ## Troubleshooting
+
+  ### Error: `pg_config executable not found` or `libpq-fe.h: No such file or directory`
+
+  This error message indicates that the [`libpq`](https://www.postgresql.org/docs/current/libpq.html) dependency is missing.
+
+  To resolve this, refer to the ["Dependencies" section](#dependencies) above.
+```
+
+## `keywords`
+
+Array of keywords associated with the plugin, for search and classification purposes.
+
+```yaml
+keywords:
+- api
+- meltano_sdk
+```
+
+## `requires`
+
+Other plugins that this plugin depends on.
+
+Example:
+
+```yaml
+requires:
+  files:
+  - name: files-airflow
+    variant: meltano
+```
+
+### `requires.<plugin_type>.name`
+
+### `requires.<plugin_type>.variant`
+
+## `commands`
```

Comment:
@edgarrmondragon - Can you also add a section for `comands.<command_name>.container_spec` and its children?

Reply:
Content can be borrows from here: https://docs.meltano.com/concepts/project#containerized-commands

And probably a cross-link from that section to this reference would also be helpful. Thanks! 👍

Reply:
@aaronsteers I've moved the doc to `/reference` and linked the sections

Reply:
Thanks, @edgarrmondragon !